### PR TITLE
Releasing v3.1.3

### PR DESCRIPTION
**Pull request by nkallima**
This release introduces the following changes/improvements:
- Fix for the `chlhash.c` and `dsmhash.c` concurrent hash-tables. 
- Fix for the functionality of `synchFastRandomRange32` and `synchFastRandomRange` provided by `fastrand.c`.
- Improvements for the benchmarking infrastructure of hash-tables.
- Adding code for validating the data-structures provided by `clhhash.c` and `dsmhash.c`.
- Increasing code coverage by `codecov`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
+[![check-build](https://github.com/nkallima/sim-universal-construction/actions/workflows/check-build.yml/badge.svg)](https://github.com/nkallima/sim-universal-construction/actions/workflows/check-build.yml) [![validate](https://github.com/nkallima/sim-universal-construction/actions/workflows/validate.yml/badge.svg)](https://github.com/nkallima/sim-universal-construction/actions/workflows/validate.yml) [![codecov](https://codecov.io/gh/nkallima/sim-universal-construction/branch/main/graph/badge.svg?token=1V8A6BOABM)](https://codecov.io/gh/nkallima/sim-universal-construction) [![status](https://joss.theoj.org/papers/07bf35ba1bd72c38cc8076fee6864409/status.svg)](https://joss.theoj.org/papers/07bf35ba1bd72c38cc8076fee6864409)
```
Comment (Spiros-N-Agathos) [inline README.md:1]: what caused this change?

File: libconcurrent/concurrent/clhhash.c [hunk 2]
```diff
@@ -76,7 +77,7 @@ static inline RetVal serialOperations(void *h, ArgVal dummy_arg, int pid) {
             if (top != prev)
                 prev->next = cur->next;
             else
-                top->next = cur->next;
+                cells[arg.cell].ptr = cur->next;
```
Comment (Spiros-N-Agathos) [inline libconcurrent/concurrent/clhhash.c:80]: was that a bug? How come and tests didn't expose this?

Comment (Spiros-N-Agathos) [reply libconcurrent/concurrent/clhhash.c:80]: EDIT: I just saw that those structures were not tested.

**Review by Spiros-N-Agathos — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (nkallima) [reply README.md:1]: Maybe an update from codecov.io?

**Pull request merged**